### DISCUSSION TITLE
Deadlinks taxon_go_triggers taxon_union_terms

### DISCUSSION THREAD

**Discussion by davidbio** [geneontology/helpdesk#458] [Q&A] [answered]
Hi GO-Team,

I am looking for 2 files, their links are dead.

ftp://ftp.geneontology.org/pub/go/quality_control/annotation_checks/taxon_checks/taxon_go_triggers.obo
ftp://ftp.geneontology.org/pub/go/quality_control/annotation_checks/taxon_checks/taxon_union_terms.obo

They are used to implement:
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-530

Can someone point me in the right direction?

**Comment by balhoff** [comment, marked as answer]
Hi @davidbio our ontology source files as well as the release process have changed significantly since then. The FTP site is no longer operating. Now you can obtain release files by date from here: http://release.geneontology.org

Furthermore for the latest version of any ontology file, it's best to use the OBO library persistent URL. For example, for the main GO release, you can obtain it here:
- http://purl.obolibrary.org/obo/go.obo

Taxon constraints are maintained using a different set of files now, but they are all included in the version of the ontology called "go-plus". This is an OWL file containing many more axioms than the core release, referencing various external terminologies. The standard release is in RDF/XML, here:
- http://purl.obolibrary.org/obo/go/extensions/go-plus.owl

But there is also a JSON version which isn't quite as complete:
- http://purl.obolibrary.org/obo/go/extensions/go-plus.json

If you aren't using an RDF or OWL library, then that version might be easiest to extract taxon relationships from, which look like this:

```json
{
      "sub" : "http://purl.obolibrary.org/obo/GO_0048102",
      "pred" : "http://purl.obolibrary.org/obo/RO_0002160",
      "obj" : "http://purl.obolibrary.org/obo/NCBITaxon_2759"
}
```

You can find information about the taxon union terms in there as well:

```json
{
      "sub" : "http://purl.obolibrary.org/obo/GO_0019812",
      "pred" : "http://purl.obolibrary.org/obo/RO_0002160",
      "obj" : "http://purl.obolibrary.org/obo/NCBITaxon_Union_0000004"
}
```

The definitions for the taxon union terms are included in `go-plus.owl`, but you can also find an OBO version here: 
- http://purl.obolibrary.org/obo/go/imports/go-taxon-groupings.obo

Here's a bit more information on ontology file downloads: https://geneontology.org/docs/download-ontology/

Also, computing the full set of taxon constraints that are in effect for a particular term requires applying an OWL reasoner to go-plus. But I'm currently in the middle of creating a fully precomputed taxon constraints file that will be available in future releases: https://github.com/geneontology/go-ontology/issues/19759

**Comment by kltm** [comment]
@davidbio I'm assuming that the source of this query is coming from the https://geneontology.org/docs/taxon-constraints/ page? The links there look to be a little off; I'm repairing them now. (Tagging @suzialeksander )

**Comment by davidbio** [comment]
@balhoff 
Thank you for the thorough response. I will check the resources you suggested and try to update the pipeline to use the new resources.

@kltm 
I don't know where we found these links originally, I just saw that they are referenced in our pipeline and don't work anymore.
